src/AccountServices.js

```diff
@@ -0,0 +1,147 @@
+var OAuth      = require('oauth');
+var TwitterHandler = require('./TwitterHandler.js').TwitterHandler;
+
+
+
+/**
+ * Construct a AccountServices.
+ * This class is a handler for conversation between users and the bridge bot to
+ * link accounts together
+ *
+ * @class
+ * @extends {external:TwitterHandler}
+ *
+ * @param  {matrix-appservice-bridge.Bridge}   bridge
+ * @param  app_auth OAuth authentication information
+ * @param  app_auth.consumer_key Twitter consumer key
+ * @param  app_auth.consumer_secret Twitter consumer secret
+ * @param  {TwitterDB}       storage
+ * @param  {MatrixTwitter}   twitter
+ */
+var AccountServices = function (bridge, app_auth, storage, twitter) {
+  TwitterHandler.call(this,bridge,null,"service");
+  this._app_auth = app_auth;
+  this._storage = storage;
+  this._twitter = twitter;
+  this._oauth = new OAuth.OAuth(
+    'https://api.twitter.com/oauth/request_token',
+    'https://api.twitter.com/oauth/access_token',
+    this._app_auth.consumer_key,
+    this._app_auth.consumer_secret,
+    '1.0A',
+    "oob",
+    'HMAC-SHA1'
+  );
+}
+
+AccountServices.prototype.processInvite = function (event, request, context){
+  var intent = this._bridge.getIntent();
+  var event = request.getData();
+  intent.join(event.room_id).then( () => {
+    var rroom = new RemoteRoom("service_"+event.sender);
+    rroom.set("twitter_type", "service");
+    this._bridge.getRoomStore().linkRooms(context.rooms.matrix,rroom);
+    intent.sendMessage(event.room_id,{"body":"This bot can help you link/unlink your twitter account with the Matrix Twitter Bridge. If this was not your intention, please kick the bot.","msgtype":"m.text"});
+    //Add the room to the list.
+  });
+};
+
+AccountServices.prototype.processMessage = function (event, request, context){
+  var intent = this._bridge.getIntent();
+  if(event.content.body == "link account"){
+    log.info("Handler.AccountServices",event.sender + " is requesting a twitter account link.");
+    this._oauth_getUrl(event,event.sender).then( (url) =>{
+      intent.sendMessage(event.room_id,{"body":"Go to "+url+" to receive your PIN, and then type it in below.","msgtype":"m.text"});
+    }).catch(err => {
+      log.error("Handler.AccountServices",err[0],err[1]);
+      intent.sendMessage(event.room_id,{"body":"We are unable to process your request at this time.","msgtype":"m.text"});
+    });
+  }
+  else if(event.content.body == "unlink account"){
+    this._storage.remove_client_data(event.sender);
+    this._storage.remove_timeline_room(event.sender);
+    this._twitter.detach_user_stream(event.sender);
+    intent.sendMessage(event.room_id,{"body":"Your account (if it was linked) is now unlinked from Matrix.","msgtype":"m.text"});
+  }
+  else if(isNumber(event.content.body)){
+    this._storage.get_client_data(event.sender).then((client_data) => {
+      var pin = event.content.body;
+      log.info("Handler.AccountServices","User sent a pin in to auth with.");
```

Comment:
Not a helpful log message. Which user? Is it safe to log the PIN? If yes, log that too.